web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/ImageServerV2ServiceTest.java

```diff
@@ -0,0 +1,133 @@
+    public void setup() throws Exception {
+        closeable = openMocks(this);
+        service = new ImageServerV2Service();
+        outputStream = new ByteArrayOutputStream();
+
+        // Configure the service
+        service.setImageServerProxyBasePath(IMAGE_SERVER_BASE_PATH);
+        service.setBasePath(BASE_PATH);
+
+        TestHelpers.setField(service, "httpClient", httpClient);
+    }
+
+    @AfterEach
+    void closeService() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void getMetadata_Success() throws Exception {
+        String jsonContent = "{\"@context\":\"http://iiif.io/api/image/2/context.json\"," +
+                "\"@id\":\"http://example.org/iiif/test\",\"protocol\":\"http://iiif.io/api/image\"," +
+                "\"width\":600,\"height\":400,\"sizes\":[{\"width\":150,\"height\":100},{\"width\":300," +
+                "\"height\":200},{\"width\":600,\"height\":400}],\"tiles\":[{\"width\":512," +
+                "\"scaleFactors\":[1,2,4,8,16]}],\"profile\":[\"http://iiif.io/api/image/2/level2.json\"," +
+                "\"http://iiif.io/api/image/2/level1.json\"]}";
+
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_OK);
+
+        StringEntity stringEntity = new StringEntity(jsonContent, ContentType.APPLICATION_JSON);
+        when(httpResponse.getEntity()).thenReturn(stringEntity);
+        when(httpClient.execute(any(HttpGet.class))).thenReturn(httpResponse);
+
+        service.getMetadata(TEST_PID, outputStream, servletResponse);
+
+        verify(httpClient).execute(httpGetCaptor.capture());
+        HttpGet httpGet = httpGetCaptor.getValue();
+        assertEquals(IMAGE_SERVER_BASE_PATH + TEST_ENCODED_ID + TEST_PID + ".jp2/info.json", httpGet.getURI().toString());
+
+        verify(servletResponse).setHeader("Content-Type", "application/json");
+        verify(servletResponse).setHeader("content-disposition", "inline");
+
+        // Verify content was written to output stream
+        String outputContent = outputStream.toString();
+        // We can't do an exact match because the content is transformed, but we can verify it's not empty
+        assert(!outputContent.isEmpty());
+    }
+
+    @Test
+    public void getMetadata_ClientAbort() throws Exception {
+        when(httpResponse.getStatusLine()).thenReturn(statusLine);
+        when(statusLine.getStatusCode()).thenReturn(HttpStatus.SC_OK);
+        when(httpClient.execute(any(HttpGet.class))).thenReturn(httpResponse);
+
+        when(httpResponse.getEntity()).thenReturn(httpEntity);
+        when(httpEntity.getContent()).thenThrow(new IOException("Connection reset"));
+
+        service.getMetadata(TEST_PID, outputStream, servletResponse);
+
+        verify(httpClient).execute(any(HttpGet.class));
+    }
+
+    @Test
+    public void setConnectionManager() {
+        service.setHttpClientConnectionManager(connectionManager);
```

Comment:
what is this testing? just that it doesn't error?

Reply:
Yeah pretty much, it's a setter that does a bunch of stuff, but it's not stuff that is needed in a unit test, and there isn't good visibility into it's outcomes.